scripts/min_glibc_version.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# This scripts determines the minimum GLIBC version required for a dynamic library.
+
+set -e -o pipefail
+
+# List the dynamic symbols of the library
+# Note: `ldd` does not work reliably when inspecting cross-compiled ARM binaries on x86_64
+objdump -T "$1" \
+  |
+  # Select only the GLIBC symbols
+  grep -o '(GLIBC_[0-9][0-9]*.*)' \
+  |
+  # Version-sort the unique symbols, descending
+  sort -r -V | uniq \
+  |
+  # Print out just the version number
+  sed -n 's/.*GLIBC_\([0-9.]*\)).*/\1/p' \
+  |
+  # Finally, print out the highest version
```

Comment:
not sure I understand why is this picking the highest version. Shouldn't it be the other way around?

Reply:
As the title in the script says:

```
This scripts determines the minimum GLIBC version required for a dynamic library.
```
 
To visualize: Let's say your binary links to symbols that requires GLIBC 2.17, GLIBC 2.20, and GLIBC 2.5. The minimum supported version of GLIBC for that binary is 2.20 - if you attempted to load it on a system with a lower GLIBC, it would fail, since it would not have an expected symbol from the higher, required version.